atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TestTransactionManagerImpl.java

```diff
@@ -108,10 +112,37 @@ public Transaction commitAndStartNewTransaction(Transaction tx) {
     }
 
     @Override
-    public Transaction createNewTransaction() {
-        Map<TableReference, ConflictHandler> conflictHandlersWithOverrides = new HashMap<>();
-        conflictHandlersWithOverrides.putAll(conflictDetectionManager.getCachedValues());
-        conflictHandlersWithOverrides.putAll(conflictHandlerOverrides);
+    protected SnapshotTransaction createTransaction(
+            long immutableTimestamp,
+            Supplier<Long> startTimestampSupplier,
+            LockToken immutableTsLock,
+            PreCommitCondition condition) {
+        Map<TableReference, ConflictHandler> conflictHandlersWithOverrides = getConflictHandlerWithOverrides();
+        return new SnapshotTransaction(
+                keyValueService,
+                timelockService,
+                transactionService,
+                cleaner,
+                startTimestampSupplier,
+                TestConflictDetectionManagers.createWithStaticConflictDetection(conflictHandlersWithOverrides),
+                sweepStrategyManager,
+                immutableTimestamp,
+                Optional.of(immutableTsLock),
+                condition,
+                constraintModeSupplier.get(),
+                cleaner.getTransactionReadTimeoutMillis(),
+                TransactionReadSentinelBehavior.THROW_EXCEPTION,
+                allowHiddenTableAccess,
+                timestampValidationReadCache,
+                lockAcquireTimeoutMs.get(),
+                getRangesExecutor,
+                defaultGetRangesConcurrency,
+                sweepQueueWriter);
+    }
+
+    @Override
+    public SnapshotTransaction createNewTransaction() {
```

Comment:
Not sure I understand why we changed the return type of this one from `Transaction` to `SnapshotTransaction` - do we actually use the properties of a snapshot-transaction in the test?